### PR TITLE
Avoid compiler casting warnings by assigning to variables of the same type where possible

### PR DESCRIPTION
**Pull request by AtariDreams**
This saves instructions that are related to casting as well as compiler warnings.

### CODE REVIEW

**Comment by millert** [comment]
Are you using gcc's -Wconversion or something similar to find these?  Some of the changes are OK but others would break the plugin or library ABI which is not OK.  Also, some of the non-trivial changes introduce bugs.

**Comment by AtariDreams** [comment]
@millert Fixed!

File: lib/util/strtoid.c [hunk 1]
```diff
@@ -80,7 +80,7 @@ sudo_strtoidx_v1(const char *p, const char *sep, char **endp, const char **errst
 	 * Disallow id -1 (UINT_MAX), which means "no change"
 	 * and check for a valid separator (if specified).
 	 */
-	if (ret == (id_t)-1 || ret == (id_t)UINT_MAX || !valid_separator(p, ep, sep)) {
+	if (ret == (id_t)-1 || !valid_separator(p, ep, sep)) {
```
Comment (millert) [inline lib/util/strtoid.c:83]: This needs to stay.  On some systems id_t is a signed 64-bit integer so both -1 and UINT_MAX are possible.

File: plugins/sudoers/getdate.c [hunk 2]
```diff
@@ -2345,7 +2345,7 @@ difftm(struct tm *a, struct tm *b)
 {
   int ay = a->tm_year + (TM_YEAR_ORIGIN - 1);
   int by = b->tm_year + (TM_YEAR_ORIGIN - 1);
-  int days = (
```
Comment (millert) [inline plugins/sudoers/getdate.c:2348]: This change needs to be made to getdate.y as well.

**Review by millert — CHANGES_REQUESTED** [review]

File: lib/util/explicit_bzero.c [hunk 3]
```diff
@@ -70,7 +70,7 @@ sudo_explicit_bzero(void *v, size_t n)
 
     /* Updating through a volatile pointer should not be optimized away. */
     while (n--)
-	*s++ = '\0';
+	*s++ = 0;
```
[the remark this answers is not in the document]
Comment (AtariDreams) [reply lib/util/explicit_bzero.c:73]: char casted to unsigned char

**Comment by AtariDreams** [comment]
@millert Done!

File: lib/eventlog/eventlog.c [hunk 4]
```diff
@@ -317,12 +317,11 @@ exec_mailer(int pipein)
...
-    if ((p = strtok_r(mflags, " \t", &last))) {
-	do {
-	    argv[i] = p;
-	} while (++i < MAX_MAILFLAGS && (p = strtok_r(NULL, " \t", &last)));
+    for (p = strtok_r(mflags, " \t", &last); p != NULL;
+            p = strtok_r(NULL, " \t", &last)) {
+        if (i < MAX_MAILFLAGS)
+            argv[i++] = p;
     }
-    argv[i] = NULL;
```
Comment (millert) [inline lib/eventlog/eventlog.c:325]: The manual NULL termination of the array is still needed here.

File: src/limits.c [hunk 5]
```diff
@@ -708,7 +708,7 @@ serialize_rlimits(char **info, size_t info_max)
     }
     debug_return_int(nstored);
 oom:
-    while (nstored--)
-	free(info[nstored]);
+    while (nstored)
+    free(info[--nstored]);
```
Comment (millert) [inline src/limits.c:712]: Please indent the free() call here.

**Review by millert — CHANGES_REQUESTED** [review]
Just two minor issues.

**Comment by AtariDreams** [comment]
@millert Done!

**Pull request merged**